src/assets/js/ledyer-payments.js

```diff
@@ -0,0 +1,274 @@
+                    } else {
+                        console.warn( "AJAX request succeeded, but the Woo order was not created.", data )
+                        throw "SubmitOrder failed"
+                    }
+                } catch ( err ) {
+                    console.error( err )
+                    if ( data.messages ) {
+                        // Strip HTML code from messages.
+                        const messages = data.messages.replace( /<\/?[^>]+(>|$)/g, "" )
+
+                        logToFile( "Checkout error | " + messages, "error" )
+                        submitOrderFail( "submitOrder", messages )
+                    } else {
+                        logToFile( "Checkout error | No message", "error" )
+                        submitOrderFail( "submitOrder", "Checkout error" )
+                    }
+                }
+            },
+            error: ( data ) => {
+                try {
+                    logToFile( "AJAX error | " + JSON.stringify( data ), "error" )
+                } catch ( e ) {
+                    logToFile( "AJAX error | Failed to parse error message.", "error" )
+                }
+                submitOrderFail( "AJAX", "Something went wrong, please try again." )
+            },
+        } )
+    }
+
+    let field = $( "#billing_company_number_field" ).detach()
+    const moveCompanyNumberField = () => {
+        if ( gatewayParams.companyNumberPlacement === "billing_form" ) {
+            if ( isActiveGateway() ) {
+                $( "#billing_company_number_field" ).detach()
+                field.insertAfter( "#billing_company_field" )
+            } else {
+                field = $( "#billing_company_number_field" ).detach()
+            }
+        }
+    }
+
+    $( "body" ).on( "click", "input#place_order, button#place_order", ( e ) => {
+        if ( ! isActiveGateway() ) {
+            return
+        }
+
+        const organizationNumber = $( "#billing_company_number" ).val().trim()
+        if ( organizationNumber.length === 0 ) {
+            printNotice( gatewayParams.i18n.companyNumberMissing )
+            return false
+        }
+
+        submitOrder( e )
+    } )
+
+    $( document ).ready( () => {
+        // If "billing_form", remove the field from the payment_form and insert it after the company name field. Otherwise, if it is "payment_form", leave as-is.
+        if ( gatewayParams.companyNumberPlacement === "billing_form" ) {
+            if ( isActiveGateway() ) {
+                $( "#billing_company_number_field" ).detach().insertAfter( "#billing_company_field" )
+            }
+
+            // Required whenever the customer changes payment method.
+            $( "body" ).on( "change", 'input[name="payment_method"]', moveCompanyNumberField )
+            // Required when the checkout is initially loaded, and Ledyer is the chosen gateway.
+            $( "body" ).on( "updated_checkout", moveCompanyNumberField )
+        }
+    } )
+} )
```

Comment:
When rewriting this to a object instead, these can be registered in the init function as listeners with separate functions for them.